app/products/calls/screens/call_screen/index.ts

```diff
@@ -3,24 +3,27 @@
 
 import withObservables from '@nozbe/with-observables';
 import {combineLatest, of as of$} from 'rxjs';
-import {switchMap} from 'rxjs/operators';
+import {distinctUntilChanged, switchMap} from 'rxjs/operators';
 
 import CallScreen from '@calls/screens/call_screen/call_screen';
 import {observeCurrentCall} from '@calls/state';
 import {CallParticipant} from '@calls/types/calls';
 import DatabaseManager from '@database/manager';
-import {observeTeammateNameDisplay, observeUsersById} from '@queries/servers/user';
+import {observeTeammateNameDisplay, queryUsersById} from '@queries/servers/user';
+
+import type UserModel from '@typings/database/models/servers/user';
 
 const enhanced = withObservables([], () => {
     const currentCall = observeCurrentCall();
     const database = currentCall.pipe(
         switchMap((call) => of$(call ? call.serverUrl : '')),
+        distinctUntilChanged(),
         switchMap((url) => of$(DatabaseManager.serverDatabases[url]?.database)),
     );
     const participantsDict = combineLatest([database, currentCall]).pipe(
-        switchMap(([db, call]) => (db && call ? observeUsersById(db, Object.keys(call.participants)) : of$([])).pipe(
+        switchMap(([db, call]) => (db && call ? queryUsersById(db, Object.keys(call.participants)).observeWithColumns(['nickname', 'username', 'firstname', 'lastname', 'first_name', 'last_name', 'lastPictureUpdate', 'last_picture_update']) : of$([])).pipe(
```

Comment:
```suggestion
        switchMap(([db, call]) => (db && call ? queryUsersById(db, Object.keys(call.participants)).observeWithColumns(['nickname', 'username', 'first_name', 'last_name', 'last_picture_update']) : of$([])).pipe(
```

suggestion is to only add the name of the database columns

Reply:
Gotcha -- I made the change. I wasn't sure if we had to observe the `UserModel` fields as well... 
Actually, looking at `queryUsersById`, it looks like it's getting `UserModel`, so shouldn't we be observing the `firstname`, `lastname` and `lastPictureUpdate` columns? (mentioning so this question doesn't get lost: @enahum )

Reply:
Yes it returns the model, but you observe the database columns by their name, not the model property names.

The model will map the database field to its property, if you tske a look to the actual model you'll see they apply a decorator called `@field` that does the mapping